src/PatientPage/CheckupNot.tsx

```diff
@@ -0,0 +1,185 @@
+import type { CheckupNotUnreadQuery as CheckupNotUnreadQueryType } from './__generated__/CheckupNotUnreadQuery.graphql';
+import type { ErrorBoundary } from 'react-error-boundary';
+import type { PreloadedQuery } from 'react-relay';
+
+import CheckupNotUnreadQuery from './__generated__/CheckupNotUnreadQuery.graphql';
+
+import React, {
+    useCallback,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
+import { Center, Grid } from '@chakra-ui/react';
+
+import { useQueryLoader } from 'react-relay';
+import { usePreloadedQuery } from 'react-relay';
+import { graphql } from 'babel-plugin-relay/macro';
+
+import Suspense from 'Suspense';
+import NotBox from './NotBox';
+
+import { useAuthenticatedToken } from 'authentication';
+import { useBackendURL } from 'config';
+import { usePatientId } from 'user';
+
+type LoadedProps = {
+  data: PreloadedQuery<CheckupNotUnreadQueryType>,
+}
+
+function LoadedCheckupNot(props: LoadedProps) {
+    const data = usePreloadedQuery(
+        graphql`
+        query CheckupNotUnreadQuery {
+          me {
+            firstname
+            lastname
+            patientProfile {
+                address {
+                    city
+                }
+                unreadCheckups {
+                  id
+                  services
+                }
+              }
+          }
+      }
+      `,
+        props.data,
+    );
+    
+    return (
+        <Center>
+            <Grid gap={6} w={600}>
+                { data.me?.patientProfile?.unreadCheckups !== undefined &&
+                data.me?.patientProfile?.unreadCheckups.length > 0 &&
+                data.me?.patientProfile?.unreadCheckups.map(checkup => {
+                    return (
+                        <NotBox
+                            city={data.me?.patientProfile?.address?.city !== undefined ?
+                                data.me?.patientProfile?.address?.city : ''}
+                            key={checkup.id}
+                            keyV={checkup.id}
+                            name={`${data.me?.firstname} ${data.me?.lastname}`}
+                            service={checkup.services[0]}
+                        />
+                    );
+                })
+                }
+            </Grid>
+        </Center>
+    );
+}
+
+function CheckupNot() {
+    const url = useBackendURL('graphql');
+    const accessToken = useAuthenticatedToken();
+    const patientId = usePatientId();
+
+    const [refreshCheckups, setRefreshCheckups] = useState(true);
+    
+    const fetchRec = useCallback(
```

Comment:
So you generate the recommendations... and then the frontend passes those to a mutation...
This still seems like creating them from a read operation again. The only difference is that the frontend is now passing the data back.

My question is do we absolutely have to store these recommendations? Or could we just store them whenever they're manipulated for the first time. Like if I set the recommendation as read, only then do we create an entry in the db that the user has read the recommendation

Reply:
These recommendations might not be exactly like checkup appointments and that's ok. For one they wouldn't have an id. They can be completely different types

Reply:
But still I need a way to remember that they marked them as read or used them.
I am not sure if recommendations and reminders are different. Otherwise why ti use the reminders?

"Like if I set the recommendation as read, only then do we create an entry in the db that the user has read the recommendation" We could do that as well but I dont thik it changes the whole concept

Reply:
I dont know if you have something different in mind but I couldnt think a different way to implement it